F03-CreateEquipmentService

Azenord opened the pull request:
Create Equipment Service
Change EquipmentRepository and Equipment model

Azenord committed:
CreateEquipmentService

Create Equipment Service
Change EquipmentRepository and Equipment model

Excome commented on src/main/kotlin/ru/nngasu/finalqualifyingproject/model/Equipment.kt:
```diff
@@ -18,4 +16,20 @@ data class Equipment(
     @GeneratedValue(strategy = GenerationType.AUTO)
     var id: Long = 0
+
+    fun getEquipmentName(): String {
```
Геттеры не нужны, в Котлине доступ к полям осуществляется напрямую (e.g. equipment_obg.name)

Excome commented on src/main/kotlin/ru/nngasu/finalqualifyingproject/model/Equipment.kt:
```diff
@@ -18,4 +16,20 @@ data class Equipment(
+        return this.name
+    }
+
+    fun getEquipmentCount(): Int {
```
Геттеры не нужны, в Котлине доступ к полям осуществляется напрямую (e.g. equipment_obg.name)

Excome commented on src/main/kotlin/ru/nngasu/finalqualifyingproject/model/Equipment.kt:
```diff
@@ -18,4 +16,20 @@ data class Equipment(
+        return this.count
+    }
+
+    fun getEquipmentType(): String {
```
Геттеры не нужны, в Котлине доступ к полям осуществляется напрямую (e.g. equipment_obg.name)

Excome commented on src/main/kotlin/ru/nngasu/finalqualifyingproject/model/Equipment.kt:
```diff
@@ -18,4 +16,20 @@ data class Equipment(
+        return this.type
+    }
+
+    fun getEquipmentDescription(): String{
```
Геттеры не нужны, в Котлине доступ к полям осуществляется напрямую (e.g. equipment_obg.name)

Excome commented on src/main/kotlin/ru/nngasu/finalqualifyingproject/model/Equipment.kt:
```diff
@@ -1,14 +1,12 @@
 
 /**
-@author Peshekhonov Maksim
+@author Lalykin Kirill
```
Аннотация @author в java-doc'e остается за тем кто создал класс, а не кто последний его поменял

Excome commented on src/main/kotlin/ru/nngasu/finalqualifyingproject/repository/EquipmentRepository.kt:
```diff
@@ -3,10 +3,13 @@ package ru.nngasu.finalqualifyingproject.repository
 
 /**
-@author Peshekhonov Maksim
+@author Lalykin Kirill
```
Аннотация @author в java-doc'e остается за тем кто создал класс, а не кто последний его поменял

Excome commented on src/main/kotlin/ru/nngasu/finalqualifyingproject/service/EquipmentService.kt:
```diff
@@ -0,0 +1,42 @@
+    fun createEquipment(name: String, type: String, count: Int, description: String): Equipment {
+        var equipmentFromDb = equipmentRepository.findEquipmentByName(name = name)
+        if (equipmentFromDb != null)
+            throw UserException("Equipment with name '${name}' already exist", UserError.EMAIL_IS_ALREADY_USED) // с исключением не понятно немного, не разобрался. Поставил UserError пока что.
```
Для ентити Equipment нужно создать свой тип Exception (e.g. EquipmentException) и свой тип Error(e.g. EquipmentError), по аналогии с UserException и UserError и использовать их здесь

Excome commented on src/main/kotlin/ru/nngasu/finalqualifyingproject/service/EquipmentService.kt:
```diff
@@ -0,0 +1,42 @@
+        var equipmentFromDb = equipmentRepository.findEquipmentByName(name = name)
+        if (equipmentFromDb != null)
+            throw UserException("Equipment with name '${name}' already exist", UserError.EMAIL_IS_ALREADY_USED) // с исключением не понятно немного, не разобрался. Поставил UserError пока что.
+        equipmentFromDb = equipmentRepository.findEquipmentByType(type = type)
```
Зачем поиск и проверка оборудования по типу в базе? Достаточно лишь поиска по имени т.к. тип оборудования (e.g Laptop, Printer и т.д.) могут повторяться. У нас же может быть разные модели ноутбуков по имени, но тип то у них будет одинаковый

Excome commented on src/main/kotlin/ru/nngasu/finalqualifyingproject/service/EquipmentService.kt:
```diff
@@ -0,0 +1,42 @@
+
+        return equipmentRepository.save(Equipment(name = name, type = type, count = count, description = description))
+    }
+    fun getEquipmentByEquipmentName(name: String): Equipment {
```
переименуй метод в getEquipmentByName

Excome commented on src/main/kotlin/ru/nngasu/finalqualifyingproject/service/EquipmentService.kt:
```diff
@@ -0,0 +1,42 @@
+            ?: throw UserException("Equipment with type '${type} not found!'", UserError.USER_NOT_FOUND) // Ну ты понял
+    }
+
+    fun saveEquipment(equipment: Equipment): Equipment {
```
Не нужный метод, уже есть метод createEquipment для создания оборудования в базе

Azenord committed:
CreateEquipmentException

Create equipment exception and exception error
remove any function in equipment.kt and rename author
small fix equipment repositry and rename author
fix exceprion in equipment service and small fix
